src/plugins/data/public/ui/query_editor/query_editor.tsx

```diff
@@ -295,12 +296,17 @@ export default class QueryEditorUI extends Component<Props, State> {
     return {
       suggestions:
         suggestions && suggestions.length > 0
-          ? suggestions.map((s: QuerySuggestion) => ({
-              label: s.text,
-              kind: s.type as monaco.languages.CompletionItemKind,
-              insertText: s.insertText ?? s.text,
-              range,
-            }))
+          ? suggestions
+              .filter((s) => 'detail' in s) // remove suggestion not of type MonacoCompatible
```

Comment:
what is detail?

Reply:
detail is a field in a possible suggestion that is a label on what that suggestion is. for example, if we're suggesting a field of type geo_point, it will read `Field: geo_point` next to the suggestion. the reason it's being used here is as a type guard since detail is a monaco specific suggestion field